rfcs/0010-postmortem-principles.md

```diff
@@ -0,0 +1,57 @@
+---
+created-on: 2021-04-27
+freshness: 2021-04-27
+owner: @maxneuvians
+status: proposal
+tags:
+  - postmortem
+  - incident response
+---
+
+[La version française suit.](#résumé)
+
+# CDS Principles for running incident response and doing postmortems
+
+## Summary
+Things go wrong. Software breaks. Partnerships aren’t smooth. Delivery takes longer than expected. Sometimes, things go right. In all cases, we should take the time to look back on what happened, how it happened and what we can do better in the future.
+
+Postmortems are a tool for determining what happened, documenting the timelines and discovering what went right and what went wrong. They help us understand how a given outcome occurred and, if necessary, provide the action items to ensure that outcome doesn’t happen again in the future.
+
+## Motivation
+We need to align as an organisation on the value of postmortems - what they give us, why they are worth investing in, and what the consequences of ignoring outcomes are.
+
+## Proposal
+
+### People will make mistakes
+It is completely unreasonable to expect humans to never make a mistake, otherwise they would never learn. When people make mistakes, we assume they acted with the best intentions based on the information they had at the time. A postmortem needs to honestly and objectively examine the circumstances that led to the fault so we can find the true root cause(s) and mitigate them.
+
+### People need to feel psychologically safe
+Witnessing the public shaming and blaming of a co-worker for mistakes is a surefire way of reducing the confidence and productivity of others. Future incidences will most likely be covered up and the organisation as a whole will suffer from lost productivity and low morale.
+
+### Incidents are an opportunity to learn and reflect
+In the broadest sense, an incident is any unplanned work you encounter in your day-to-day operations. Good judgement is required to know when unplanned events require an incident response, as seemingly innocuous events may have terrifying root causes. This will come with experience running incident responses, which means it is better to err on running more than running less.
+
+### Timelines are important
+It is important to keep an up-to-date timeline of events during an incident as it will be the only moment in time when the information is fresh in your mind. Humans naturally collapse details as time goes by and important nuances will be lost. Similarly, it will give team members joining the incident later on a good overview of where the process is at.
+
+### Action items need ownership and accountability
+We can have the best postmortem process in the world, but have it amount to nothing unless we follow up on the action items identified and hold ourselves accountable to them. It is not enough to identify them, they need to be put into backlogs and actioned with dedicated owners.
```

Comment:
I also like the idea of reviewing postmortems with other teams to highlight how things can break and how they were fixed (even if it isn't directly related to the work "you" do).  It would help build psychological safety.

Reply:
I'd love to see this happen in the weekly dev community meeting. Take 5-10 minutes and run through a postmortem, what happened, how we caught it, what AIs were listed and how they were handled.

Reply:
Thank you!